backend/src/routes/api/cluster-settings/clusterSettingsUtils.ts

```diff
@@ -81,20 +115,66 @@ export const getClusterSettings = async (
   };
+  const dashConfig = getDashboardConfig();
   try {
     const segmentEnabledRes = await coreV1Api.readNamespacedConfigMap(segmentKeyCfg, namespace);
     clusterSettings.userTrackingEnabled = segmentEnabledRes.body.data.segmentKeyEnabled === 'true';
   } catch (e) {
     fastify.log.error('Error retrieving segment key enabled: ' + e.toString());
   }
-  try {
-    const jupyterhubCfgResponse = await coreV1Api.readNamespacedConfigMap(juypterhubCfg, namespace);
-    clusterSettings.pvcSize = Number(
-      jupyterhubCfgResponse.body.data.singleuser_pvc_size.replace('Gi', ''),
-    );
-    clusterSettings.cullerTimeout = Number(jupyterhubCfgResponse.body.data.culler_timeout);
-  } catch (e) {
-    fastify.log.error('Error retrieving cluster settings: ' + e.toString());
+  if (dashConfig.spec?.notebookController?.enabled) {
+    let nbcCfgResponse;
+    try {
+      nbcCfgResponse = await fastify.kube.coreV1Api.readNamespacedConfigMap(
+        nbcCfg,
+        fastify.kube.namespace,
+      );
+      const cullerTimeout = nbcCfgResponse.body.data['CULL_IDLE_TIME'];
+      const isEnabled = Boolean(nbcCfgResponse.body.data['ENABLE_CULLING']);
+      if (isEnabled) {
+        clusterSettings.cullerTimeout = Number(cullerTimeout) * 60; //minutes to seconds;
+      } else {
+        clusterSettings.cullerTimeout = DEFAULT_CULLER_TIMEOUT; // For backwards compatibility with jupyterhub and less changes to UI
+      }
+      clusterSettings.pvcSize = 20; //PLACEHOLDER
+    } catch (e) {
+      if (e.statusCode === 404) {
+        fastify.log.warn('Creating new notebook controller culling settings.');
+        nbcCfgResponse = await fastify.kube.coreV1Api.createNamespacedConfigMap(
+          fastify.kube.namespace,
+          defaultCullingSettings,
+        );
+        const cullerTimeout = nbcCfgResponse.body.data['CULL_IDLE_TIME'];
+        const isEnabled = Boolean(nbcCfgResponse.body.data['ENABLE_CULLING']);
+        if (isEnabled) {
+          clusterSettings.cullerTimeout = Number(cullerTimeout) * 60; //minutes to seconds;
+        } else {
+          clusterSettings.cullerTimeout = DEFAULT_CULLER_TIMEOUT; // For backwards compatibility with jupyterhub and less changes to UI
+        }
+        clusterSettings.pvcSize = 20; //PLACEHOLDER
+      } else {
+        fastify.log.error('Error getting notebook controller culling settings: ' + e.toString());
+      }
+    }
+  } else {
+    try {
+      const [pvcSize, cullerTimeout] = await readJupyterhubCfg(fastify);
+      clusterSettings.pvcSize = pvcSize;
+      clusterSettings.cullerTimeout = cullerTimeout;
+    } catch (e) {
+      fastify.log.error('Error retrieving cluster settings: ' + e.toString());
+    }
   }
+
   return clusterSettings;
 };
+
+const readJupyterhubCfg = async (fastify: KubeFastifyInstance): Promise<[number, number]> => {
+  const jupyterhubCfgResponse = await fastify.kube.coreV1Api.readNamespacedConfigMap(
+    jupyterhubCfg,
+    fastify.kube.namespace,
+  );
```

Comment:
Recommend you do a `.then` on this and normalize your data away from the kube library's response.

Eg something like:
```
.then((response) => response.body)
```

That way your variable `jupyterhubCfgResponse` can just be a `ConfigMap`.